crates/holochain_p2p/src/spawn/actor.rs

```diff
@@ -192,18 +198,176 @@ impl event::HcP2pHandler for WrapEvtSender {
     }
 }
 
-#[derive(Debug)]
+type Respond = tokio::sync::oneshot::Sender<crate::wire::WireMessage>;
+
+struct Pending {
+    this: Weak<Mutex<Self>>,
+    map: HashMap<u64, Respond>,
+}
+
+impl Pending {
+    fn register(&mut self, msg_id: u64, resp: Respond) {
+        if let Some(this) = self.this.upgrade() {
+            self.map.insert(msg_id, resp);
+            tokio::task::spawn(async move {
+                tokio::time::sleep(TIMEOUT).await;
+                let _ = this.lock().unwrap().respond(msg_id);
+            });
+        }
+    }
+
+    fn respond(&mut self, msg_id: u64) -> Option<Respond> {
+        self.map.remove(&msg_id)
+    }
+}
+
 pub(crate) struct HolochainP2pActor {
+    this: Weak<Self>,
     evt_sender: WrapEvtSender,
+    lair_client: holochain_keystore::MetaLairClient,
+    kitsune: Mutex<Option<kitsune2_api::DynKitsune>>,
+    pending: Arc<Mutex<Pending>>,
+}
+
+impl std::fmt::Debug for HolochainP2pActor {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("HolochainP2pActor").finish()
```

Comment:
Basic information worth having like presence of a K2 handle and the number of pending requests?

Reply:
Maybe we debug print this so rarely it's not worth the effort

Reply:
Yeah, don't have much of a preference.